File: integrasjon/src/main/java/no/nav/melosys/integrasjon/Fagsystem.java
package no.nav.melosys.integrasjon;

import no.nav.melosys.domain.Kodeverk;

public enum Fagsystem implements Kodeverk {

    GSAK("FS19", "GSAK"),
    // GSAK er fagsystemet vi skal bruke til dokumentarkivering, men av historiske årsaker har det feil kode i Joark
    GSAK_I_JOARK("FS22", "GSAK i Joark"),
    MELOSYS("FS38", "Melosys"); // "https://confluence.adeo.no/display/APPKAT/Applikasjons-ID"

    private String kode;
    private String beskrivelse;

    private Fagsystem(String kode, String beskrivelse) {
        this.kode = kode;
        this.beskrivelse = beskrivelse;
    }

    @Override
    public String getKode() {
        return kode;
    }

    @Override
    public String getBeskrivelse() {
        return beskrivelse;
    }
}

File: domain/src/main/kotlin/no/nav/melosys/domain/Fagsystem.kt
package no.nav.melosys.domain

import no.nav.melosys.domain.kodeverk.Kodeverk

/**
 * https://confluence.adeo.no/display/APPKAT/Applikasjons-ID
 */
enum class Fagsystem(
    private val kode: String,
    private val beskrivelse: String
) : Kodeverk {
    // GSAK brukes som referanse til sakstilhørende fagsystem, men av historiske årsaker har det feil kode i Joark.
    GSAK_I_JOARK("FS22", "GSAK i Joark"), // Bruker Gosys appID
    INTET("", ""),
    MELOSYS("FS38", "Melosys");

    override fun getKode(): String = kode
    override fun getBeskrivelse(): String = beskrivelse
}
